### PR TITLE
draco: update versions

### PR DESCRIPTION
**Pull request by KineticTheory**
+ Added versions 7.3.0 and 7.4.0.
+ Change several variants to be default TRUE since most consumers need these variants enabled (eospac, lapack, parmetis, superlu-dist). Change variant name for `+superlu_dist` to use hyphen instead of underscore.  This makes the variant name consistent with the spackage name for `superlu-dist`.
+ Clean up `depends_on` instructions and avoid specifying `type` when possible.
+ Provide patch files that are necessary for some machines (mostly Cray machines).

### CODE REVIEW

**Comment by adamjstewart** [comment]
> Change variant name for +superlu_dist to use hyphen instead of underscore. This makes the variant name consistent with the spackage name for superlu-dist.

I would recommend against this. Variants can be disabled using `~` or `-`, so variants that have a dash in their name are a little ambiguous (does `-superlu-dist` mean `~superlu~dist` or `~superlu_dist`?)

**Comment by adamjstewart** [comment]
You can also add:
```python
maintainers = ['KineticTheory']
```
so we know who to ping if someone has an installation problem or wants to modify the package in a PR.

File: var/spack/repos/builtin/packages/draco/package.py [hunk 1]
```diff
@@ -63,11 +72,3 @@ def cmake_args(self):
...
         ])
         return options
-
-    @run_after('build')
-    @on_package_attributes(run_tests=True)
-    def check(self):
-        """Run ctest after building project."""
-
-        with working_dir(self.build_directory):
-            ctest()
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/draco/package.py]: Why was the test removed? Is `make test` equivalent?

Comment (KineticTheory) [reply var/spack/repos/builtin/packages/draco/package.py]: Correct.  It was equivalent to `make test`.

**Pull request merged**